docs/contributing/coding-style-guide.md

```diff
@@ -0,0 +1,665 @@
+
+<table>
+<thead align="center"><tr><th>Avoid 🔥</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+func copyIntoSliceAndMap(biggy []string) (a []string, b map[string]struct{})
+    b = map[string]struct{}{}
+
+    for _, item := range biggy {
+        a = append(a, item)
+        b[item] = struct{}
+    }
+}
+```
+
+</td></tr>
+</tbody></table>
+<table>
+<thead align="center"><tr><th>Better 🤓</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+func copyIntoSliceAndMap(biggy []string) (a []string, b map[string]struct{})
+    b = make(map[string]struct{}, len(biggy))
+    a = make([]string, len(biggy))
+
+    // Copy will not even work without pre-allocation.
+    copy(a, biggy)
+    for _, item := range biggy {
+        b[item] = struct{}
+    }
+}
+```
+
+</td></tr>
+</tbody></table>
+
+#### Reuse arrays
+
+To extend above point, there are cases where you don't need to allocate anything all the time. If you repeat certain operation on slices
+sequentially, it's reasonable to reuse underlying array for those. This can give quite enormous gains for critical paths.
+Unfortunately there is no way to reuse underlying array for maps.
+
+<table>
+<thead align="center"><tr><th>Avoid 🔥</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+var messages []string{}
+for _, msg := range recv {
+    messages = append(messages, msg)
+
+    if len(messages) > maxMessageLen {
+        marshalAndSend(messages)
+        // This creates new array. Previous array
+        // will be garbage collected only after
+        // some time (seconds), which
+        // can create enormous memory pressure.
+        messages = []string{}
+    }
+}
+```
+
+</td></tr>
+</tbody></table>
+<table>
+<thead align="center"><tr><th>Better 🤓</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+var messages []string{}
+for _, msg := range recv {
+    messages = append(messages, msg)
+
+    if len(messages) > maxMessageLen {
+        marshalAndSend(messages)
+        // Instead of new array, reuse
+        // the same, with the same capacity,
+        // just length equals to zero.
+        messages = messages[:0]
+    }
+}
+```
+
+</td></tr>
+</tbody></table>
+
+### Readability
+
+#### Only Two Ways of Formatting Functions/Methods
+
+Prefer function/method definitions with arguments in a single line. If it's too wide, put each argument on a new line.
+
+<table>
+<thead align="center"><tr><th>Avoid 🔥</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+func function(argument1 int, argument2 string,
+    argument3 time.Duration, argument4 someType,
+    argument5 float64, argument6 time.Time,
+) (ret int, err error) {
+```
+
+</td></tr>
+</tbody></table>
+<table>
+<thead align="center"><tr><th>Better 🤓</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+func function(
+    argument1 int,
+    argument2 string,
+    argument3 time.Duration,
+    argument4 someType,
+    argument5 float64,
+    argument6 time.Time,
+) (ret int, err error)
+```
+
+</td></tr>
+</tbody></table>
+
+#### Control Structure: Avoid `else`
```

Comment:
This one is commonly referred to as "Early returns".

How about?
```suggestion
#### Control Structure: Prefer early returns (Avoid `else`)
```

Reply:
:+1: